NEWS.md

```diff
@@ -4,6 +4,10 @@
 
 - Fix internal test skipping functions for MacOS R. 
 
+### Bug Fixes
+
+- Previously if an optimizer fails `h_record_all_output` will return a `list()` or `try-error`. Now only `try-error` will be returned.
```

Comment:
In principle correct, but for the NEWS let's keep it at the user level, and describe which problem for the user is now fixed